pywemo/ouimeaux_device/__init__.py

```diff
@@ -236,6 +256,293 @@ def explain(self):
+            if not access_point.strip() or '|' not in access_point:
+                continue
+            LOG.debug('found AP: %s', access_point)
+            if access_point.startswith(f'{ssid}|'):
+                selected_ap = access_point
+                LOG.info('selecting AP: %s', selected_ap)
+                break
+
+        if selected_ap is None:
+            raise APNotFound(f'AP with SSID {ssid} not found.  Try again.')
+
+        # get some information about the access point
+        columns = selected_ap.split('|')
+        channel = columns[1].strip()
+        auth_mode, encryption_method = columns[-1].strip().split('/')
+        LOG.debug('AP channel: %s', channel)
+        LOG.debug('AP authorization mode(s): %s', auth_mode)
+        LOG.debug('AP encryption method: %s', encryption_method)
+
+        # check if the encryption type is supported by this script
+        supported_encryptions = {'NONE', 'AES'}
+        if encryption_method not in supported_encryptions:
+            raise SetupException(
+                f'Encryption {encryption_method} not currently supported.  '
+                f'Supported encryptions are: {",".join(supported_encryptions)}'
+            )
+
+        # try to connect the device to the selected network
+        if encryption_method == 'NONE':
+            LOG.debug('selected network has no encryption (password ignored)')
+            auth_mode = 'OPEN'
+            encrypted_password = ''
+        else:
+            # get the meta information of the device and encrypt the password
+            metainfo = self.get_service('metainfo').GetMetaInfo()['MetaInfo']
+            encrypted_password = self.encrypt_aes128(password, metainfo)
+
+        # optionally make multiple connection attempts
+        start_time = time.time()
+        for attempt in range(connection_attempts):
+            LOG.info('sending connection request (try %s)', attempt + 1)
+            # success rate on the first attempt is much higher if the
+            # ConnectHomeNetwork command is sent twice (not sure why!)
+            for _ in range(2):
+                result = wifisetup.ConnectHomeNetwork(
+                    ssid=ssid,
+                    auth=auth_mode,
+                    password=encrypted_password,
+                    encrypt=encryption_method,
+                    channel=channel,
+                )
+                time.sleep(0.10)
+            try:
+                status = result['PairingStatus']
+            except KeyError:
+                # print entire dictionary if PairingStatus doesn't exist
+                status = result
+            LOG.info('pairing status: %s', result)
+
+            timeout_start = time.time()
+            LOG.info('starting status checks (%s second timeout)', timeout)
+            status = None
+            while time.time() - timeout_start < timeout and status != '1':
+                time.sleep(1.5)
+                status = wifisetup.GetNetworkStatus()['NetworkStatus']
```

Comment:
So it looks like the WeMoMaker has already disconnected at this point - and even if I remove the sleep - it has already disconnected. 

So perhaps wrap a try block around this and handle the fallout?

Can make a suggestion if that would be helpful.

In my testing BTW the code always worked, just reported failure!